### PR TITLE
Updated database to the latest Sabbat sets. Changed the ids to match …

### PR DESCRIPTION
**Pull request by GiottoVerducci**
…the universal ids of the cards and removed the autogenerated ids.

### CODE REVIEW

**Comment by franciscojunior** [comment]
Great work, @GiottoVerducci ! Thank you very much! Database import is a very important task when new cards arrive and sometimes, because of the utf-8 handling and double quotes, can be very tricky too. I remember I had to make some imports a lot of times to get it right. :)

Got the update up and running! I had to make some changes though:

1.  Add some lines to DatabaseHelper.java:

```
// This is the migration to update card listings to Sabbat Sets expansion
static Migration MIGRATION_7_8 = new ReplaceCryptLibraryCardsMigration(7, 8);
```
and

`.addMigrations(MIGRATION_7_8)`


Those changes are needed because Room library checks when the database version has changed and requires a migration procedure. In this case, the procedure is just to replace the cards. In the future we may have some other more complex changes.

2. Add some null checks in the entity constructors to get empty strings: 
`this.name = name == null ? "" : name;`
`this.text = text == null ? "" : text;`

and so on.


This was caused by a little difference in the imported database: where there are null values now, I used to have empty strings. But with this little change, everything goes back to normal.  I also can change the code base to take in consideration that there could be null values instead of empty strings.


Would you mind add the migration lines to your pull request so I can merge it? If you can't do that right now, I can merge your work and add the lines separately.

About the entity null values, we can change the migration database to contain empty strings instead of null values or we can change the code to take those null values in consideration instead of empty strings.

About the database import, would you mind to add the procedure you used to create the database? There are some import scripts in the project (on sql folder) which may need some adjustments. 

Thank you again for your support and help!

**Comment by GiottoVerducci** [comment]
1/ I've made the change for the migration on the Pull Request. I left
migration from 6 to 7, I suppose it's meant to be incremental.

2/ The easiest way to handle values as before would be to change the import
SQL query to replace null by empty strings:

database.execSQL("insert into CryptCard(`uid`, `Name`, `Type`, `Clan`,
`Advanced`, `Group`, `Capacity`, `Disciplines`, `Text`, `SetRarity`,
`Artist`)  select `Id`, `Name`, `Type`, *coalesce*(`Clan`,''),
*coalesce*(`Adv`,''),
`Group`, `Capacity`, `Disciplines`, `CardText`, `Set`, `Artist` from
updatedb.crypt");
database.execSQL("insert into LibraryCard(`uid`, `Name`, `Type`, `Clan`,
`Disciplines`, `PoolCost`, `BloodCost`, `ConvictionCost`, `Text`,
`SetRarity`, `Artist`)  select `Id`, `Name`, `Type`, *coalesce*(`Clan`,''),
*coalesce*(`Discipline`,''), *coalesce*(`PoolCost`,''),
*coalesce*(`BloodCost`,''),
*coalesce*(`ConvictionCost`,''), `CardText`, `Set`, `Artist` from
updatedb.library");

I did that change in my Pull Request.

3/ As for the database import procedure, I didn't use any SQL script. I did
it simply using DB Browser for SQL Lite <https://sqlitebrowser.org/dl/>

I added the how-to the README.md file

4/ I've updated the README.md to mention the Dark Pact. It's the way WW
allows users to use their IP. You should add the Dark Pact logo in the app
(eg., about menu), or when the app is launched on a splash screen.

**Comment by franciscojunior** [comment]
Thank you very much for the changes, @GiottoVerducci ! I think it is almost ready for merge. I just made a small comment about the CCP Hf on the copyright change.

File: README.md [hunk 1]
```diff
@@ -26,10 +28,33 @@ The 2.0 BETA version doens't have all the functionality of the 1.x version yet.
...
+11. Check the PK checkbox for the line Id and click on OK
+12. Repeat steps 6-11 for the other table
+13. Tools > Compact Database > select Main and click on OK
+14. File > Close Database.
 ## Copyright notice
 
-Jyhad™, Vampire: The Eternal Struggle™ and all game symbols are trademarks of Wizards of the Coast, Inc. and CCP hf. White Wolf, 
-World of Darkness and all related terms are trademarks of CCP hf. This product is not published nor endorsed by CCP hf.
+Portions of the materials are the copyrights and trademarks of White Wolf Entertainment AB, and are used with permission. All rights reserved. For more information please visit white-wolf.com.
+
```
[the remark this answers is not in the document]
Comment (GiottoVerducci) [reply README.md]: Correct. See https://www.pcgamer.com/paradox-buys-white-wolf-world-of-darkness-vampire-the-masquerade-from-ccp/

[the remark this answers is not in the document]
Comment (franciscojunior) [reply README.md]: Great! Merging now...

**Pull request merged**